src/Symfony/Component/Lock/Store/CombinedStore.php

```
@@ -0,0 +1,160 @@
+    private $quorum;
+
+    /**
+     * @param StoreInterface[] $stores The decorated stores
+     * @param QuorumInterface  $quorum
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $stores, QuorumInterface $quorum)
+    {
+        foreach ($stores as $store) {
+            if (!$store instanceof StoreInterface) {
+                throw new InvalidArgumentException(
+                    sprintf('The store must implement "%s". Got "%s".', StoreInterface::class, get_class($store))
+                );
+            }
+        }
+
+        $this->stores = $stores;
+        $this->quorum = $quorum;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(Key $key)
+    {
+        $successCount = 0;
+        $failureCount = 0;
+        $storesCount = count($this->stores);
+
+        foreach ($this->stores as $store) {
+            try {
+                $store->save($key);
+                ++$successCount;
+            } catch (\Exception $e) {
+                ++$failureCount;
+            }
+
+            if (!$this->quorum->canBeMet($failureCount, $storesCount)) {
+                break;
+            }
+        }
+
+        if ($this->quorum->isMet($successCount, $storesCount)) {
+            return;
+        }
+
+        // clean up potential locks
+        $this->delete($key);
+
+        throw new LockConflictedException();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function expire(Key $key, $ttl)
+    {
+        $successCount = 0;
+        $failureCount = 0;
+        $storesCount = count($this->stores);
+
+        foreach ($this->stores as $store) {
+            if (!$store instanceof ExpirableStoreInterface) {
+                ++$successCount;
+                continue;
+            }
+
+            try {
+                $store->expire($key, $ttl);
+                ++$successCount;
+            } catch (\Exception $e) {
+                ++$failureCount;
+            }
```

I don't think we should catch \Error here. If a "fatal error" occured, we should let it bubble to the developper